### PR TITLE
Disable application tabs in the frontend if backing endpoint is not available

### PR DESCRIPTION
**Pull request by jochenwierum**
The property `spring.boot.admin.routes.endpoints` lists every application tab in the frontend. However, if some endpoints are removed, the tabs are still shown but throw an error, if they are accessed. This is the case for:

 * environment
 * metrics
 * threads
 * details (if metrics are not available)

The endpoint 'jmx' is hidden if jolokia is disabled completely.

In our project, we only allow access to the bean 'ch.qos.logback.classic.jmx.JMXConfigurator' to configure logging at runtime (the log-tab is shown correctly), but the rest of the JMX beans cannot be accessed. This can be tested using the following configuration:

```
<?xml version="1.0" encoding="utf-8"?>
<restrict>
    <commands>
        <command>search</command>
    </commands>

    <allow>
        <mbean>
            <name>ch.qos.logback.classic:Name=default,Type=ch.qos.logback.classic.jmx.JMXConfigurator</name>
            <operation>*</operation>
            <attribute>*</attribute>
        </mbean>
    </allow>
</restrict>
```

This causes the jmx-tab to be shown, but it only displays a warning. So I suggest to hide the jmx tab if the beans cannot be listed.

To summarize, this PR...
 
 * ... hides the tab "environment" if the env endpoint is disabled
 * ... hides the tabs "metrics" and "details" if the metrics endpoint is disabled
 * ... hides the "threads" tab if the threads endpoint is disabled
 * ... hides the jmx tab if jolokia is not availabe or listing of beans is disabled

### CODE REVIEW

**Comment by joshiste** [comment]
looks good so far. I'll hope to find the time to merge it this weekend.

**Comment by joshiste** [comment]
Merged with b9b97677ae7b4fb36b06d6b0b7d01af7be24633b.

I did remove your changes from the details-module. The button is always shown for the details, so marking the endpoint as unavailable has no effect. What I did instead (with a8355c92238026f8821aa19f11493ba33988deeb) is to hide the sections from the details view which rely on the metrics.

**Comment by jochenwierum** [comment]
> I did remove your changes from the details-module. The button is always shown for the details, so marking the module as unavailable has no effect. What I did instead (with a8355c9) is to hide the sections from the details view which rely on the metrics.

Great! Thank you very much! I really like your work :)
